src/stdlib_32_bit_water_hashes.fypp

```diff
@@ -0,0 +1,282 @@
+!!
+!! The `unlicense` reads as follows:
+!!   This is free and unencumbered software released into the public domain.
+!!
+!!   Anyone is free to copy, modify, publish, use, compile, sell, or
+!!   distribute this software, either in source code form or as a compiled
+!!   binary, for any purpose, commercial or non-commercial, and by any
+!!   means.
+!!
+!!   In jurisdictions that recognize copyright laws, the author or authors
+!!   of this software dedicate any and all copyright interest in the
+!!   software to the public domain. We make this dedication for the benefit
+!!   of the public at large and to the detriment of our heirs and
+!!   successors. We intend this dedication to be an overt act of
+!!   relinquishment in perpetuity of all present and future rights to this
+!!   software under copyright law.
+!!
+!!   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+!!   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+!!   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+!!   IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+!!   OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+!!   ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+!!   OTHER DEALINGS IN THE SOFTWARE.
+!!
+!!   For more information, please refer to <http://unlicense.org>
+!!
+!! `WATER_HASH` is distributed as part of the `stdlib_32_bit_hash_functions.f90`
+!! module and its `stdlib_32_bit_water_hashes.f90` submodule with the Fortran
+!! Standard Library at URL: https://github.com/fortran-lang/stdlib.
+!! The Fortran Standard Library, including this code, is distributed under the
+!! MIT License as described in the `LICENSE` file distributed with the library.
+!! `WATER_HASH` differs from `waterhash.h` not only in its use of Fortran,
+!! but also in its use of signed two's complement arithmetic in contrast to
+!! the unsigned arithmetic of Ettinger and Wang Yi, and in making some of the
+!! uses of `TRANSFER` endian dependent, in an attempt to make the quality of
+!! the hash endian independent. The use of signed arithmetic may change with
+!! the planned introduction of the unsigned BITS datatype in what is currently
+!! known as Fortran 202X.
+!!
+!! To be useful this code must be processed by a processor that implements two
+!! Fortran 2008 extensions to Fortran 2003: submodules, and 64 bit (`INT64`)
+!! integers. The processor must also use two's complement integers
+!! (all Fortran 95+ processors use two's complement arithmetic) with
+!! wrap around overflow at runtime and for BOZ constants. The latest releases
+!! of the following processors are known to implement the required Fortran
+!! 2008 extensions and default to runtime wrap around overflow: FLANG,
+!! gfortran, ifort, and NAG Fortran. Older versions of gfortran will require
+!! the compiler flag, `-fno-range-check`, to ensure wrap around semantics
```

Comment:
Which versions of gfortran require the flag? I think stdlib might only support gfortran 9+ -- if those versions do not need it, then I would lean toward removing it.

While perhaps not a big deal, it could lead to complications (e.g. if one compiles stdlib with user-specified flags, this could be missed, so might require documentation in the main README or similar).

Reply:
Version 9 and earlier require the flag.